src/js/actions/libraries.js

```diff
@@ -147,9 +349,72 @@ define(function (require, exports) {
         });
     };
     createLayerFromElement.reads = [locks.JS_LIBRARIES, locks.JS_DOC];
-    createLayerFromElement.writes = [locks.JS_LIBRARIES, locks.JS_DOC];
+    createLayerFromElement.writes = [locks.JS_LIBRARIES, locks.JS_DOC, locks.PS_DOC];
 
     /**
+     * Applies the given layer style element to the active layers
+     *  - Gets the path from primary representation of the asset
+     *  - Passes it to PS call ApplyLayerStyle
+     *
+     * @param {AdobeLibraryElement} element [description]
+     *
+     * @return {Promise}
+     */
+    var applyLayerStyle = function (element) {
+        var appStore = this.flux.store("application"),
+            currentDocument = appStore.getCurrentDocument();
+
+        if (!currentDocument) {
+            return Promise.resolve();
+        }
+
+        var representation = element.getPrimaryRepresentation();
+
+        return Promise.fromNode(function (cb) {
+            representation.getContentPath(cb);
+        }).bind(this).then(function (path) {
+            var layerRef = layerEffectAdapter.referenceBy.current,
+                placeObj = layerEffectAdapter.applyLayerStyleFile(layerRef, path);
+
+            return descriptor.playObject(placeObj);
+        }).then(function () {
+            // FIXME: This can be more optimistic
+            return this.transfer(layerActions.resetLayers, currentDocument, currentDocument.layers.selected);
```

Comment:
Does this need to include the ancestors as well? I don't think so, but just making sure.